pkg/config/constants/constants.go

```diff
@@ -159,4 +160,12 @@ const (
 
 	// DataplaneMode namespace label for determining ambient mesh behavior
 	DataplaneMode = "istio.io/dataplane-mode"
+
+	// AmbientRedirection specifies whether a pod has ambient redirection (to ztunnel) configured.
+	AmbientRedirection = "ambient.istio.io/redirection"
+	// AmbientRedirectionEnabled indicates redirection is configured. This is set by the CNI when it
+	// actually sets up redirection, rather than by the user.
+	AmbientRedirectionEnabled = "enabled"
```

Comment:
IIUC, from the comment the annotation is set by CNI? If so, we increased calling k8s apiserver in each node. Is there any workaround not calling k8s

Reply:
It's 1 Pod update call for each pod creation. A new pod in k8s already causes >10 API calls, probably much more. Adding annotations in CNI is done by calico, cilium, etc as well